clang/test/SemaSYCL/spec_const_accesors.cpp

```diff
@@ -0,0 +1,24 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -fsyntax-only -I %S/Inputs %s
+// The test checks that Clang doesn't crash if multiple accessors are used
+// together with specialization constants in single kernel.
```

Comment:
I would rename the test to make it's purpose more clear - e.g. `spec_const_multiple_accessors_crash.cpp`.

Reply:
Did this test previously crash in the frontend?  This tests needs some sort of verify/check lines or something...

Reply:
Yes, it crashed before.

Reply:
In the process of simplification I figured out that to reproduce the crash it is enough just to capture a spec constant *before* an accessor. So I have:
 * updated the commit message and comments
 * renamed the test to `spec_const_and_accesor_crash.cpp`
 * and added some `CHECK:` lines.

Reply:
Great! I'm OK with this, though I suspect it'll need to be rebased once the #1517 is committed (which I imagine will be momentarily).  I STILL very much want the test (I'm Ok with it as is if @bader is) and want to make sure that we didn't re-make this problem in the re-implementation.